app/helpers/foreman_ansible/hosts_helper_extensions.rb

```diff
@@ -13,22 +13,34 @@ def ansible_roles_present?(host)
         host.inherited_ansible_roles.present?
     end
 
-    def ansible_roles_button(host)
-      link_to(
-        icon_text('play', ' ' + _('Ansible roles'), :kind => 'fa'),
-        play_roles_host_path(:id => host.id),
-        :id => :ansible_roles_button,
-        :class => 'btn btn-default',
-        :'data-no-turbolink' => true
-      )
+    def ad_hoc_role_button
+      [link_to_function(
+        icon_text('play', ' ' + _('Ad hoc Ansible role'), :kind => 'fa'),
+        'show_ad_hoc_role_modal()',
```

Comment:
I am getting `Uncaught ReferenceError: show_ad_hoc_role_modal is not defined` when clicking the button